### ISSUE TITLE
Should unknown lint tools be a warning?

### ISSUE THREAD

**Issue by ehuss** [rust-lang/cargo#12496]
### Problem

When a new tool is introduced to the lints table, past versions of cargo will generate an error about an unknown tool. This can make it difficult to adopt new tools since it sets a floor on which Rust version is allowed. I'm wondering if it makes sense to treat unknown tools as a warning. Other tables in the manifest usually treat these as warnings (like `[package.foo]` is a warning).

### Steps

Have a manifest with something like:

```toml
[package]
name = "foo"
version = "0.1.0"

[lints.new-cool-tool]
new-lint = "warn"
```

This generates an error:

```
error: failed to parse manifest at `/Users/eric/Temp/z20/Cargo.toml`

Caused by:
  unsupported `new-cool-tool` in `[lints]`, must be one of rust, clippy, rustdoc
```

### Possible Solution(s)

Generate a warning instead of an error for unknown tools.

### Version

```text
cargo 1.73.0-nightly (d78bbf4bd 2023-08-03)
release: 1.73.0-nightly
commit-hash: d78bbf4bde3c6b95caca7512f537c6f9721426ff
commit-date: 2023-08-03
host: aarch64-apple-darwin
libgit2: 1.6.4 (sys:0.17.2 vendored)
libcurl: 7.88.1 (sys:0.4.65+curl-8.2.1 system ssl:(SecureTransport) LibreSSL/3.3.6)
ssl: OpenSSL 1.1.1u  30 May 2023
os: Mac OS 13.4.1 [64-bit]
```

**Comment by epage** [comment]
Giving this a try with what rustc does:
```rust
#!/usr/bin/env -S cargo -Zscript

#![warn(new_cool_tool::new_lint = "warn")]

fn main() {}
```
```
$ ./cargo-12496.rs
warning: `package.edition` is unspecified, defaulting to `2021`
   Compiling cargo-12496 v0.0.0 (/home/epage/src/personal/dump)
error[E0452]: malformed lint attribute input
 --> cargo-12496.rs:3:9
  |
3 | #![warn(new_cool_tool::new_lint = "warn")]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ bad attribute argument

For more information about this error, try `rustc --explain E0452`.
error: could not compile `cargo-12496` (bin "cargo-12496") due to previous error
```

**Comment by epage** [comment]
The reason I made this an error, instead of a warning, was to prevent abuse of the `[lints]` table by arbitrary tools.

**Comment by weihanglo** [comment]
If rustc gives an error, maybe cargo would be better follow.

**Comment by epage** [comment]
While there is room for improvement, I believe this specific issue was resolved by #13833.
